### ISSUE TITLE
Add option to use the same deep level when wrapping a thing.

### ISSUE THREAD

**Issue by joente** [thingsdb/ThingsDB#301]
**Is your feature request related to a problem? Please describe.**
Especially with *wrap-only* types you want to tell ThingsDB to include a *thing* or a *list* / *set* of *things* and not explicitly set the deep level to another value than the default 1.

**Describe the solution you'd like**
Allow a character, for example `&` to prefix a definition. This character is only applicable when wrapping a type and tells ThingsDB to use the *same* deep level for that property. 

> For example

```javascript
new_type('User');
set_type('User', {
    name: 'str',
    friend: '&User?'
});

iris = User{
    name: 'Iris',
    friend: User{
        name: 'Cato'
    }
};

iris.wrap();
```

> Return value in JSON format

```json
{
    "friend": {
        "friend": null,
        "name": "Cato"
    },
    "name": "Iris"
}
```

**Describe alternatives you've considered**
As an alternative, we could introduce *anonymous* types which could have the same effect and do not allow the creation of separate types to configure complex nesting. This option would arguably be even better but this option is much harder to implement and is not excluded by this solution. Thus, in a future version we can still choose to implement *anonymous* types.

**Comment by joente** [comment]
Implemented in **v1.3.0**

**Comment by joente** [comment]
See documentation: https://docs.thingsdb.net/v1/overview/type/#same-deep-level
